diff --git a/source/zombie/iso/IsoGridSquare.java b/source/zombie/iso/IsoGridSquare.java
index 0e9a23d..21cc093 100644
--- a/source/zombie/iso/IsoGridSquare.java
+++ b/source/zombie/iso/IsoGridSquare.java
@@ -6670,6 +6670,10 @@ public final class IsoGridSquare {
                 if (!ServerOptions.instance.safehouseAllowFire.getValue()) {
                     return;
                 }
+
+                if (GameServer.server && this.getBuilding() != null && SafeHouse.isSafeHouse(this, null, false) == null) {
+                    return;
+                }
             }
 
             for (int i = 0; i < this.specialObjects.size(); i++) {
diff --git a/source/zombie/iso/objects/IsoFire.java b/source/zombie/iso/objects/IsoFire.java
index 5cecfb2..16b146b 100644
--- a/source/zombie/iso/objects/IsoFire.java
+++ b/source/zombie/iso/objects/IsoFire.java
@@ -279,8 +279,8 @@ public class IsoFire extends IsoObject {
                 ? false
                 : smoke
                     || !GameServer.server && !GameClient.client
-                    || SafeHouse.getSafeHouse(gridSquare) == null
-                    || ServerOptions.instance.safehouseAllowFire.getValue();
+                    || (gridSquare.getBuilding() == null && SafeHouse.getSafeHouse(gridSquare) == null)
+                    || (SafeHouse.getSafeHouse(gridSquare) != null && ServerOptions.instance.safehouseAllowFire.getValue());
         }
     }
 
