diff --git a/source/zombie/Lua/LuaEventManager.java b/source/zombie/Lua/LuaEventManager.java
index 53194fb..5a18c96 100644
--- a/source/zombie/Lua/LuaEventManager.java
+++ b/source/zombie/Lua/LuaEventManager.java
@@ -835,6 +835,8 @@ public final class LuaEventManager implements JavaFunction {
         AddEvent("OnSteamRefreshInternetServers");
         AddEvent("OptionControllerButtonStyleChanged");
         AddEvent("OptionGamepadBindingPresetChanged");
+        AddEvent("onPlayerFullyConnected");
+        AddEvent("onPlayerDisconnected");
     }
 
     public static void clear() {
diff --git a/source/zombie/core/raknet/UdpConnection.java b/source/zombie/core/raknet/UdpConnection.java
index ffdef67..4ebf076 100644
--- a/source/zombie/core/raknet/UdpConnection.java
+++ b/source/zombie/core/raknet/UdpConnection.java
@@ -33,6 +33,7 @@ import zombie.network.server.EventManager;
 import zombie.network.statistics.data.NetworkStatistic;
 import zombie.util.StringUtils;
 import zombie.vehicles.BaseVehicle;
+import zombie.Lua.LuaEventManager;
 
 public class UdpConnection extends PacketsCache implements IConnection {
     private static final long PLAYER_SAVE_INTERVAL = 180000L;
@@ -391,6 +392,7 @@ public class UdpConnection extends PacketsCache implements IConnection {
         this.setConnectionTimestamp(15000L);
         ConnectionManager.log("fully-connected", "", this);
         EventManager.instance().report("[" + this.userName + "] connected to server");
+        LuaEventManager.triggerEvent("onPlayerFullyConnected", this.userName, this.steamId);
     }
 
     @Override
diff --git a/source/zombie/network/GameServer.java b/source/zombie/network/GameServer.java
index ab1b0bc..e434358 100644
--- a/source/zombie/network/GameServer.java
+++ b/source/zombie/network/GameServer.java
@@ -2655,6 +2655,7 @@ public class GameServer {
             LoggerManager.getLogger("user")
                 .write(connection.getIDStr() + " \"" + player.getUsername() + "\" disconnected player " + LoggerManager.getPlayerCoords(player));
             SteamGameServer.RemovePlayer(player);
+            LuaEventManager.triggerEvent("onPlayerDisconnected", player.getUsername(), player.getSteamID());
         }
     }
 
