diff --git a/source/zombie/iso/worldgen/zones/ZoneGenerator.java b/source/zombie/iso/worldgen/zones/ZoneGenerator.java
index 12d6768..f790ed6 100644
--- a/source/zombie/iso/worldgen/zones/ZoneGenerator.java
+++ b/source/zombie/iso/worldgen/zones/ZoneGenerator.java
@@ -12,6 +12,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.Set;
+import java.util.stream.Collectors;
 import java.util.Map.Entry;
 import se.krka.kahlua.vm.KahluaTable;
 import zombie.ChunkMapFilenames;
@@ -75,10 +76,10 @@ public class ZoneGenerator {
                     Set<Integer> requestedZones;
                     try {
                         int[] samples = this.map.getZones(minChunkX + x, minChunkY + y, BiomeMap.Type.ZONE);
-                        requestedZones = (Set<Integer>)Arrays.stream(samples)
+                        requestedZones = Arrays.stream(samples)
                             .distinct()
                             .boxed()
-                            .collect(HashSet::new, HashSet::add, AbstractCollection::addAll);
+                            .collect(Collectors.toCollection(HashSet::new));
                     } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
                         requestedZones = Set.of(255);
                     }
@@ -102,7 +103,7 @@ public class ZoneGenerator {
             for (Entry<Integer, Boolean[]> entry : foraging.entrySet()) {
                 Integer key = (Integer)entry.getKey();
                 if (this.map.getZoneName(key) == null) {
-                    DebugLog.log("Zone " + key + " not found in ZONE_MAP");
+                    DebugLog.log("Zone " + key + " not found in ZONE_MAP [" + metaCellX + "x" + metaCellY + "]");
                 } else {
                     Boolean[] hasForaging = (Boolean[])entry.getValue();
                     int yoffset = 0;
